### PR TITLE
improve findBestMatchExpression performance

### PR DESCRIPTION
**Pull request by AnshulMalik**
### Summary of Changes

* Find expressions in the line with binary search
* Iterate from back to find the best match with early return

### CODE REVIEW

File: src/utils/ast.js [hunk 1]
```diff
@@ -13,32 +13,114 @@ import type {
...
+      left = mid;
+    } else if (result > 0) {
+      right = mid - 1;
+    } else {
+      left = mid + 1;
+    }
+  }
+
+  return left;
+}
```
[the remark this answers is not in the document]
Comment (AnshulMalik) [reply src/utils/ast.js]: I think merging them would make them less readable, but moving them to different file is a good choice.

File: src/utils/ast.js [hunk 2]
```diff
@@ -13,32 +13,114 @@ import type {
...
+  array: ArraySymbolDeclarations,
+  location: ColumnPosition
+) {
+  // Returns the declarations which are in same line of `location`
+  const l = lowerBound(array, location, isOverlapping);
+  const u = upperBound(array, location, isOverlapping);
+  return array.slice(l, u + 1);
+}
+
 export function findBestMatchExpression(
```
[the remark this answers is not in the document]
Comment (AnshulMalik) [reply src/utils/ast.js]: Yep more tests!

**Comment by jasonLaster** [comment]
@AnshulMalik have you measured this in perf.html or the chrome devtools performance tool?

**Comment by AnshulMalik** [comment]
Nope, I haven't yet measured the performance, will soon do.

**Comment by darkwing** [comment]
Hello @AnshulMalik!  I'm going to close this for now -- feel free to pick up when you get a chance.  Thank you!

**Pull request closed without merging**